src/MaaCore/Task/Miscellaneous/StartGameTaskPlugin.cpp

```diff
@@ -9,7 +9,49 @@ bool StartGameTaskPlugin::_run()
     if (m_client_type.empty()) {
         return false;
     }
-    return ctrler()->start_game(m_client_type);
+
+    // check for android version, because it leads to different magic values
+    // https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/8966#issuecomment-2094369694
+    if (ctrler()->get_version() > 8) {
+        do {
+            if (!ctrler()->start_game(m_client_type)) {
+                return false;
+            }
+
+            // https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/8961#issue-2277568882
+            // 167: magic value needs to be >164 but <172 (as that's max)
+            if (ctrler()->get_pipe_data_size() > 167) {
+                return true;
+            }
+
+            sleep(1500);
+
+        } while (true);
+    }
+    else {
+        do {
+            if (!ctrler()->start_game(m_client_type)) {
+                return false;
+            }
+
+            // 145: needs to be >85 but <153 (as that's max)
+            // magic value lowered because of different android version
+            if (ctrler()->get_pipe_data_size() > 145) {
+                // As there's no way to know (in Android < 8) if Ark has correctly started
+                // we run the command a few more times to be sure
+                for (auto _ = 3; _--;) {
+                    sleep(1500);
+                    if (!ctrler()->start_game(m_client_type)) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+
+            sleep(1500);
+
+        } while (true);
+    }
```

Comment:
This should be a problem on mac, and it seems to conflict with restarting the ADB on the UI, etc., which will report a touch exception if the ADB is killed when connecting again.

Reply:
But #9074 doesn't seem to be MAC and I checked the log but it's too small and doesn't show the Arknights startup debug. (What I modified)

Reply:
d023 reports TouchModeNotAvailable but doesn't flashback at startup, the other two versions just go away.

Reply:
Starting AK should be the last step, after the connection, the minitouch push, and the minitouch test. Without logs I can't see exactly where the issue is. If it's in the Starting intent for AK.
If it was, the log would be full of spam of this: 

<details>
<summary>Details</summary>

```py
[2024-05-12 03:01:21.473][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 99 ms , stdout size: 75 , socket size: 0
[2024-05-12 03:01:21.473][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }


[2024-05-12 03:01:21.473][TRC][Px25c4][Tx0fc8] ready to sleep 1500
[2024-05-12 03:01:22.974][TRC][Px25c4][Tx0fc8] end of sleep 1500
[2024-05-12 03:01:23.082][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 109 ms , stdout size: 75 , socket size: 0
[2024-05-12 03:01:23.082][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }


[2024-05-12 03:01:23.082][TRC][Px25c4][Tx0fc8] ready to sleep 1500
[2024-05-12 03:01:24.583][TRC][Px25c4][Tx0fc8] end of sleep 1500
[2024-05-12 03:01:24.686][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 103 ms , stdout size: 75 , socket size: 0
[2024-05-12 03:01:24.686][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }


[2024-05-12 03:01:24.686][TRC][Px25c4][Tx0fc8] ready to sleep 1500
[2024-05-12 03:01:26.187][TRC][Px25c4][Tx0fc8] end of sleep 1500
[2024-05-12 03:01:26.295][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 107 ms , stdout size: 75 , socket size: 0
[2024-05-12 03:01:26.295][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }


[2024-05-12 03:01:26.295][TRC][Px25c4][Tx0fc8] ready to sleep 1500
[2024-05-12 03:01:27.795][TRC][Px25c4][Tx0fc8] end of sleep 1500
[2024-05-12 03:01:27.899][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 104 ms , stdout size: 75 , socket size: 0
[2024-05-12 03:01:27.899][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }


[2024-05-12 03:01:27.899][TRC][Px25c4][Tx0fc8] ready to sleep 1500
[2024-05-12 03:01:29.400][TRC][Px25c4][Tx0fc8] end of sleep 1500
[2024-05-12 03:01:29.502][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 102 ms , stdout size: 75 , socket size: 0
[2024-05-12 03:01:29.502][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }


[2024-05-12 03:01:29.502][TRC][Px25c4][Tx0fc8] ready to sleep 1500
[2024-05-12 03:01:31.004][TRC][Px25c4][Tx0fc8] end of sleep 1500
[2024-05-12 03:01:31.098][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 94 ms , stdout size: 75 , socket size: 0
[2024-05-12 03:01:31.098][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }


[2024-05-12 03:01:31.098][TRC][Px25c4][Tx0fc8] ready to sleep 1500
[2024-05-12 03:01:32.599][TRC][Px25c4][Tx0fc8] end of sleep 1500
[2024-05-12 03:01:32.695][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 96 ms , stdout size: 75 , socket size: 0
[2024-05-12 03:01:32.696][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }


[2024-05-12 03:01:32.696][TRC][Px25c4][Tx0fc8] ready to sleep 1500
[2024-05-12 03:01:34.197][TRC][Px25c4][Tx0fc8] end of sleep 1500
[2024-05-12 03:01:34.309][INF][Px25c4][Tx0fc8] Call ` D:\Downloads\MAA_Y\adb\adb.exe -s 127.0.0.1:16384 shell am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext ` ret 0 , cost 112 ms , stdout size: 173 , socket size: 0
[2024-05-12 03:01:34.309][TRC][Px25c4][Tx0fc8] stdout output:
Starting: Intent { cmp=com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext }

Warning: Activity not started, intent has been delivered to currently running top-most instance.

```

</details>